src/Microsoft.AspNet.Mvc.Razor/Precompilation/RazorPreCompiler.cs

```diff
@@ -291,6 +293,13 @@ private static Stream GetNonDisposableStream(Stream stream)
             return new NonDisposableStream(stream);
         }
 
+        private static string CombinePath(string root, string name)
+        {
+            // We use string.Join instead of Path.Combine here to ensure that the path
+            // separator we produce matches the one used by the CompilerCache.
+            return string.Join(PathSeparator, root, name);
```

Comment:
https://github.com/aspnet/Mvc/pull/3134/files#diff-7f27afec00025f958d6da462dfbe5c93L83 - we specify the root path as an empty string.